client/components/autocomplete2/test/index.test.js

```diff
@@ -47,12 +48,53 @@ describe('Autocomplete', () => {
     expect(component.instance().engine.get.mock.calls[0][0]).toBe('a title')
   })
 
-  it('OnSelect calls props.onSelect with selected id', () => {
+  it('OnSelect calls props.onSelect with selected id', async () => {
     const component = mount(
       <Autocomplete {...props} />
     )
-    component.instance().onSelect(results[0])
-    expect(props.onSelect.mock.calls[0][0][0]).toBe(results[0]._id)
+    await component.instance().onSelect(searchResults[0])
+    expect(props.onSelect.mock.calls[0][0][0]).toBe(searchResults[0].id)
+  })
+
+  it('Returns a custom formatted selection on select if provided', async () => {
+    const formatSelected = async (item) => {
+      try {
+        const article = new Article(item)
+        article.fetch = jest.fn().mockReturnThis()
+        return await article.fetch()
+      } catch (error) {
+        return error
+      }
+    }
```

Comment:
One little tip for mocking this is that Backbone fetches return promises, so there shouldn't be a need to wrap it in the try/catch and async/await on it. You can mock Backbone.sync instead. This might need tweaking, but something like this general pattern should work:

```
Backbone.sync = jest.fn()
const formatSelected = selected => new Article(selected).fetch()
```
I won't let this block you since what you have works, but just some food for thought.

Reply:
Much better!